mmdet/datasets/coco.py

```diff
@@ -384,18 +386,19 @@ def evaluate(self,
                 print_log(log_msg, logger=logger)
                 continue
 
-            if metric not in result_files:
-                raise KeyError(f'{metric} is not in results')
+            metric_type = 'bbox' if metric == 'f1' else metric
+            if metric_type not in result_files:
+                raise KeyError(f'{metric_type} is not in results')
             try:
-                cocoDt = cocoGt.loadRes(result_files[metric])
+                cocoDt = cocoGt.loadRes(result_files[metric_type])
```

Comment:
why did you rename `metric` to `metric_type`?

Reply:
To evaluate f1, I  need to load 'bbox' results. So, metric_type = 'bbox' if the we calculate 'f1' and, otherwise, metric_type=metric, like it was before my changes.
```metric_type = 'bbox' if metric == 'f1' else metric```  (389 line)